common/components/common/integrations/SocialMediaSignupSection.jsx

```diff
@@ -124,12 +128,23 @@ class SocialMediaSignupSection extends React.Component<Props, State> {
             (config.iconClass ? " " + config.iconClass : "");
           return (
             <div className="LogInController-socialLink">
-              <a href={"/api/login/" + app} key={app}>
+             {this.state.prevPageArgs ? (
+
+              <a href={"/api/login/" + app + "?prevPage=" +this.state.prevPage + "&prevPageArgs=" + this.state.prevPageArgs} key={app}>
+                <span style={{ color: config.iconColor }} className={iconClass}>
+                  {config.iconElement}
+                </span>
+                <span>Continue with {config.displayName}</span>
+              </a>
+              ) : (
+              <a href={"/api/login/" + app  + "?prevPage=" + this.state.prevPage} key={app}>
                 <span style={{ color: config.iconColor }} className={iconClass}>
                   {config.iconElement}
                 </span>
                 <span>Continue with {config.displayName}</span>
               </a>
+              
+             )}
             </div>
```

Comment:
The only difference between the link elements in this conditional is the url.  I would set the url in a separate variable above the return statement.

Also, you can simplify constructing the url arguments with
https://github.com/DemocracyLab/CivicTechExchange/blob/52e3996e3c71941f2eb0f7d628ba285b512bc3ec/common/components/utils/url.js#L256